times_reader/transforms.py

```diff
@@ -938,6 +938,77 @@ def name_comm_group(df):
             dataframe=comm_groups,
         )
     )
+
+    return tables
+
+
+def generate_top_ire(
+    tables: List[datatypes.EmbeddedXlTable],
+) -> List[datatypes.EmbeddedXlTable]:
+    """
+    Generate inter-regional exchange topology
+    """
+
+    veda_set_ext_reg_mapping = {"IMP": "IMPEXP", "EXP": "IMPEXP", "MIN": "MINRNW"}
+    dummy_process_cset = [["NRG", "IMPNRGZ"], ["MAT", "IMPMATZ"], ["DEM", "IMPDEMZ"]]
+    veda_process_sets = utils.single_table(tables, "VedaProcessSets").dataframe
+    com_map = utils.single_table(tables, "COM_GMAP").dataframe
+
+    ire_prc = pd.DataFrame(columns=["Region", "TechName"])
+    for table in tables:
+        if table.tag == datatypes.Tag.fi_process:
+            df = table.dataframe
+            ire_prc = pd.concat(
+                [ire_prc, df.loc[df["Sets"] == "IRE", ["Region", "TechName"]]]
+            )
+    ire_prc.drop_duplicates(keep="first", inplace=True)
+
+    internal_regions = pd.DataFrame([], columns=["Region"])
+    for table in tables:
+        if table.tag == datatypes.Tag.book_regions_map:
+            internal_regions = pd.concat(
+                [internal_regions, table.dataframe.loc[:, ["Region"]]]
+            )
+
+    # Generate inter-regional exchange topology
+    top_ire = pd.DataFrame(dummy_process_cset, columns=["Csets", "TechName"])
+    top_ire = pd.merge(top_ire, internal_regions, how="cross")
+    top_ire = pd.merge(top_ire, com_map[["Region", "Csets", "CommName"]])
+    top_ire.drop(columns=["Csets"], inplace=True)
+    top_ire["IO"] = "OUT"
+    top_ire = pd.concat([top_ire, com_map[["Region", "TechName", "CommName", "IO"]]])
+    top_ire = pd.merge(top_ire, ire_prc)
+    top_ire = pd.merge(top_ire, veda_process_sets)
+    top_ire["Region2"] = top_ire["Sets"].replace(veda_set_ext_reg_mapping)
```

Comment:
Perhaps use (Import/Export)Region or (Source/Destination)Region all the way through rather than Region/Region2/Source/Dest?

Reply:
I think the logic there is similar to #26

Reply:
Just so that I understand, the column names `Region`, `Region2`, `TechName`, etc used here are internal-only, right? If so, I think it might be helpful to have more verbose/clear names like `SourceRegion` and `DestinationRegion`.

Similarly, unless the column names on the LHS of the mapping below are specified by TIMES,
```
TOP_IRE[ALL_REG,COM,ALL_R,C,PRC] = TOP_IRE(Origin,IN,Destination,OUT,TechName)
```
I propose using more verbose names like `SRC_REG` and `DST_REG`.

Reply:
No, `TechName` is extracted later on as `PRC`. `Region` - is as elsewhere - region where the process is defined, while `Region2` is the region to which the trade process is connected. `Region` and `Region2` are not needed for `TOP-IRE`, so they are dropped once the `Origin` and `Destination` have been determined. 
Regarding `TOP_IRE[ALL_REG,COM,ALL_R,C,PRC]`, this is very close to what is used in the TIMES documentation (using `P` instead of `PRC` here would make it exact match). Keeping it this way makes it very easy to look up what exactly those indices include.